206 implementera klocka till arduino

DavidJanReyes opened the pull request:
# Pull Request

## Beskrivning
<!-- Kort beskrivning av vad denna PR gör och varför den behövs -->
Feature: Tid hämtas nu av Arduino från ESP32 med GET HTTP request. Arduino loggar nu data binärt till flash istället för att lagra String. 

- Tiden uppdateras sedan hela tiden med millis() för att ge en ungefärlig tid. Hämtning av tid från ESP32 görs sedan vid återanslutning till ESP om koppling skulle ha förlorats.
Just nu har vi inget sätt att hantera tiden för data som genereras när klockan inte har ställs in ännu. Dessa data kommer tillsvidare att ha tidsstämpel: 1970-01-01...

-  Arduino loggar nu data binärt med struct LogEntry som format. 

## Typ av ändring
<!-- Kryssa i det som passar -->
- [x] Ny funktionalitet
- [ ] Buggfix
- [x] Refaktorering
- [ ] Dokumentation
- [ ] Test

## Checklista
<!-- Kontrollera innan PR kan mergas -->
- [x] Kod kompilerar / byggs utan fel
- [ ] Enhetstester har körts och passerat
- [ ] Kod följer kodstandard/linting
- [ ] Nödvändiga ändringar i dokumentation gjorda
- [x] Branch är uppdaterad med senaste `main` / `develop`

## Testinstruktioner
<!-- Beskriv hur denna PR kan testas manuellt eller automatiskt -->
1. Töm först Arduinos flash genom att anropa logger.clearAll() i Setup() en gång och kör sedan upload. Detta är nödvändigt för att ta bort data som har lagrats i det gamla formatet.  
2. Koppla in arduino samt ESP. Notera tidsstämplar på arduinon. Notera när tid hämtas från ESP.
3.  Se till att tiden stämmer och ser ut ungefär som på bilderna nedan.

## Skärmdumpar / exempel
<!-- Lägg gärna till bilder, diagram eller logs om relevant -->
Följande visar att tid ej har hunnits hämtas:
<img width="696" height="151" alt="image" src="https://github.com/user-attachments/assets/c2b367a2-02d2-4af4-936b-b92f9e715694" />
Tiden uppdateras: 
<img width="487" height="70" alt="image" src="https://github.com/user-attachments/assets/d91d80e8-bc3c-4b9a-92f8-5fa0f39c9d11" />
<img width="712" height="70" alt="image" src="https://github.com/user-attachments/assets/481b4045-dfaf-4094-9008-1b8f00318b39" />

## Påverkan
- [ ] Backend
- [ ] Frontend
- [x] Embedded / Firmware
- [ ] Infrastruktur / CI/CD
- [ ] Annat (beskriv):

## Relaterade ärenden / tasks
<!-- Koppla PR till Jira/GitHub issues eller interna tasks -->
- Task-ID: 206
- Feature:

## Kommentarer till granskare
<!-- Speciellt att tänka på vid granskning -->
- Eventuella risker
- Om något behöver testas extra noga
- Särskilda delar av koden som är komplexa

DavidJanReyes committed:
Added HTTP GET Time request

DavidJanReyes committed:
Updated ArduinoLogger to store time
- Fixed timestamps showing wrong time

DavidJanReyes committed:
Merge remote-tracking branch 'origin/Development' into 206-Implementera-klocka-till-arduino

DavidJanReyes committed:
updated arduinologger.cpp

Maimehs commented:
Eftersom vi nu har tester i development så kan det vara bra att lägga till något eller några tester för att testa klockan också

DavidJanReyes committed:
Removed old tests & updated arduino-logger-tests

DavidJanReyes committed:
Updated with copilot suggestions
- Added casts to avoid corruption
- Changed some comments

DavidJanReyes committed:
Fixed misleading print

Maimehs reviewed (approved):
Ser bra ut!

DavidJanReyes committed:
Filter out bad timestamps & added get time at intervals
- Refactored some arduino code
- Added timeSync every 10min
- Filter out bad timestamps
